usr/src/cmd/zdb/zdb.c

```diff
@@ -2301,11 +2301,18 @@ dump_label(const char *dev)
 
 		(void) snprintf(path, sizeof (path), "%s%s", ZFS_RDISK_ROOTD,
 		    dev);
-		if ((s = strrchr(dev, 's')) == NULL || !isdigit(*(s + 1)))
+		if (((s = strrchr(dev, 's')) == NULL &&
+		    (s = strchr(dev, 'p')) == NULL) ||
+		    !isdigit(*(s + 1)))
 			(void) strlcat(path, "s0", sizeof (path));
```

Comment:
I am feeling a bit odd about just using "s0" there, right now it is the case of course. with "7446 zpool create should support efi system partition" it could be "s1" as well, but 7446 has not landed. But I wonder if we should be a bit smarter there..  in any case, this is small nit IMO and not an blocker.